client/asset/dcr/dcr.go

```diff
@@ -6156,6 +6538,14 @@ func (dcr *ExchangeWallet) monitorBlocks(ctx context.Context) {
 			if walletTip == nil {
 				// Mempool tx seen.
 				dcr.emitBalance()
+
+				go func() {
+					dcr.tipMtx.RLock()
+					tip := dcr.currentTip
+					dcr.tipMtx.RUnlock()
+					dcr.syncTxHistory(ctx, uint64(tip.height))
+				}()
```

Comment:
Why does this need to be asynchronous?

Reply:
I guess it can only be a longer running function on the initial call in `Connect`. After your rescanning PR gets merged though and this can get reset to 0, it could become a longer running function whenever the rescan completes.